### PR TITLE
Add a pair of convolutional models for CIFAR10 image classification

### PR DESCRIPTION
**Pull request by jekbradbury**
These are intended to match the networks used in the canonical PyTorch 
and Keras CIFAR10 examples, but they don't currently converge to the 
correct loss. Requires https://github.com/rxwei/DeepLearning/pull/4 but no
compiler changes.

### CODE REVIEW

File: CIFAR/data.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        .reshaped(to: [imageCount, 3, 32, 32])
+        .transposed(withPermutations: [0, 2, 3, 1]))
+
+    return (Tensor<Int32>(labelTensor), imageTensor / Float(255.0))
+}
+
+func loadTrainingBatches() -> (Tensor<Int32>, Tensor<Float>) {
+  let data = (1..<6).map { loadBatches(from: "data_batch_\($0)") }
+  return (Raw.concat(concatDim: Tensor<Int32>(0), data.map { $0.0 }),
+  Raw.concat(concatDim: Tensor<Int32>(0), data.map { $0.1 }))
```
Comment (dan-zheng) [inline CIFAR/data.swift]: Indent:
```suggestion
  return (Raw.concat(concatDim: Tensor<Int32>(0), data.map { $0.0 }),
          Raw.concat(concatDim: Tensor<Int32>(0), data.map { $0.1 }))
```

---

We should probably add a wrapper around `Raw.concat` in the stdlib (taking an array of tensors-to-concatenate). Currently there's only the binary [`Tensor.concatenated(with: Tensor)`](https://github.com/apple/swift/blob/9ffbeee115e6da724a171f4aa58b318dcea16a7a/stdlib/public/TensorFlow/Ops.swift#L687) (or [`x ++ y`](https://github.com/apple/swift/blob/9ffbeee115e6da724a171f4aa58b318dcea16a7a/stdlib/public/TensorFlow/Ops.swift#L706)).

File: CIFAR/main.swift [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import TensorFlow
+
+let batchSize: Int32 = 4
+
+let (trainingDataset, testDataset) = loadCIFAR10()
+let trainingBatches = trainingDataset.batched(Int64(batchSize))
+let testBatches = testDataset.batched(Int64(batchSize))
+
+let modeRef = ModeRef()
+var net = CIFARNet() // modeRef: modeRef)
```
Comment (dan-zheng) [inline CIFAR/main.swift]: nit: rename for consistency?
```suggestion
var model = CIFARNet() // modeRef: modeRef)
```

**Comment by jekbradbury** [comment]
Yes, differentiation works and the model trains towards convergence, although not with exactly the same training curve as the Keras example it intends to replicate. I will work on root-causing the difference.

**Comment by jekbradbury** [comment]
One more note: the Python calls we use in these scripts currently require Python 3.x. I've been compiling with `swiftc ~/swift-models/CIFAR/*.swift` and running with `LD_LIBRARY_PATH=~/miniconda3/lib/ ./main` using miniconda Python 3.7.
